build-info-extractor-docker/src/main/java/org/jfrog/build/extractor/docker/extractor/BuildDockerCreate.java

```diff
@@ -78,77 +105,125 @@ public static void main(String[] ignored) {
+        List<String> dataLines = Files.readAllLines(Paths.get(kanikoImageFile));
         if (dataLines.isEmpty()) {
-            throw new IOException("empty image file \"" + filePath + "\".");
+            throw new IOException("Couldn't read image file \"" + kanikoImageFile + "\".");
         }
         final List<ImageFileWithDigest> imageFileWithDigests =
                 dataLines.stream().map(this::getImageFileWithDigest).filter(Objects::nonNull).collect(Collectors.toList());
         if (imageFileWithDigests.size() != dataLines.size()) {
-            throw new RuntimeException("missing image-tag/sha256 in file: \"" + filePath + "\"");
+            throw new RuntimeException("missing image-tag/sha256 in file: \"" + kanikoImageFile + "\"");
         }
         return imageFileWithDigests;
     }
 
+    /**
+     * Read the file which contains the following format:
+     * {
+     * "image":"IMAGE-NAME-IN-ARTIFACTORY",
+     * "tags":["IMAGE-TAGS-IN-ARTIFACTORY"]
+     * "imageId":"sha256:SHA256-OF-THE-IMAGE-MANIFEST",
+     * "imageDigest":"sha256:SHA256-OF-THE-IMAGE",
+     * }
+     *
+     * @param jibImageFiles - Pattern matches JIB's image JSON files. For example: "*target/jib-image.json".
+     * @return the image file with names and digests extracted.
+     */
+    private List<ImageFileWithDigest> getJibImageFilesWithDigests(String jibImageFiles) {
+        ObjectMapper mapper = createMapper();
+        String baseDir = UploadSpecHelper.getWildcardBaseDir(new File(""), jibImageFiles);
+        String newPattern = UploadSpecHelper.prepareWildcardPattern(new File(""), jibImageFiles, baseDir);
+        String regexPath = PathsUtils.pathToRegExp(newPattern);
+        List<ImageFileWithDigest> imageFilesWithDigests = new ArrayList<>();
+        Path baseDirPath = Paths.get(baseDir);
+        try (Stream<Path> files = Files.find(baseDirPath,
+                Integer.MAX_VALUE,
+                (path, basicFileAttributes) -> baseDirPath.relativize(path).toString().matches(regexPath))) {
+            files.forEach(jibImageFile -> {
+                JsonNode jsonNode;
+                try {
+                    jsonNode = mapper.readTree(jibImageFile.toFile());
+                } catch (IOException e) {
+                    throw new RuntimeException("Couldn't read image file \"" + jibImageFiles + "\".");
+                }
+                JsonNode imageName = jsonNode.get("image");
+                if (imageName == null || imageName.isNull()) {
+                    throw new RuntimeException("Missing \"image\" in file: \"" + jibImageFiles + "\"");
+                }
+                JsonNode imageDigest = jsonNode.get("imageDigest");
+                if (imageDigest == null || imageDigest.isNull()) {
+                    throw new RuntimeException("Missing \"imageDigest\" in file: \"" + jibImageFiles + "\"");
+                }
+                imageFilesWithDigests.add(new ImageFileWithDigest(imageName.asText(), imageDigest.asText()));
```

Comment:
Should this be adding a image file instance for each of the tags in the file?

Reply:
Unlike Kaniko, the image file of JIB contains only one image. It is concluded from the [description](https://github.com/GoogleContainerTools/jib/blob/master/jib-maven-plugin/README.md#extended-usage) of the jib-image.json:
> The path of the image metadata json file written out during the build. Relative paths are resolved relative to the project root.

To support multiple JIB image files, we allow providing a Wildcard pattern. For example, the default pattern is: `*/target/jib-image.json`.